### PR TITLE
Add auth token requirement for trace-agent set log_level (APMSP-1204)

### PR DESCRIPTION
**Pull request by ajgajg1134**
### What does this PR do?
Requires the agent's auth token to modify the trace-agent's log level.

### Motivation
It's important that changes to log level are only done by authenticated users

### Describe how to test/QA your changes
I tested this change using a local agent where I used Fleet Management to request a remote flare, this then triggered remote config to update the log level to debug mode which it did correctly and with a valid authentication token.

### Possible Drawbacks / Trade-offs
I don't _love_ needing to pass this GetAuthToken function into the agent config but it was the most reasonable way I could determine to do this while avoiding importing pkg/config (which is very large and we don't want it in pkg/trace)

### CODE REVIEW

**Review by ichinaski — APPROVED** [review]
LGTM, just small question about a function whose use I cannot really spot

File: releasenotes/notes/traceagent-loglevel-027cc2c0e581b195.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    APM: The trace agent endpoint for changing the configured log level now requires authentication so it is only accessible to other agent processes.
```
Comment (jhgilbert) [inline releasenotes/notes/traceagent-loglevel-027cc2c0e581b195.yaml:11]:
```suggestion
    APM: The trace agent endpoint for changing the configured log level now requires authentication so it is only accessible to other Agent processes.
```

**Review by jhgilbert — APPROVED** [review]
Approved with a minor suggestion, thanks!

File: comp/trace/config/config.go [hunk 2]
```diff
@@ -133,6 +133,10 @@ func (c *cfg) SetHandler() http.Handler {
 			httpError(w, http.StatusMethodNotAllowed, fmt.Errorf("%s method not allowed, only %s", req.Method, http.MethodPost))
 			return
 		}
+		if apiutil.Validate(w, req) != nil {
+			return
+		}
+		log.Info("WE VALIDATED THE REQUEST")
```
Comment (ichinaski) [inline comp/trace/config/config.go:139]: Is this log intended?

**Review by ichinaski — APPROVED** [review]

File: comp/trace/config/config.go (hunk 2 above)
Comment (ajgajg1134) [reply comp/trace/config/config.go:139]: oops, it is not. I was using it locally to make absolutely sure my test was working as expected

**Review by ankitpatel96 — APPROVED** [review]

**Review by pducolin — APPROVED** [review]

**Review by ogaca-dd — APPROVED** [review]

**Review by gh123man — APPROVED** [review]
AML owned files LGTM

**Pull request merged**